codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/customization/S3UpdateEndpoint.java

```diff
@@ -102,6 +134,175 @@ public void writeAdditionalFiles(
         }
     }
 
+
+
+
+    /**
+     * writes BackfillAccountID Helper function for s3 api client
+     *
+     *  Generates code:
+     *  === api_client.go===
+     *  func backfillAccountID(in interface, v string) (interface{},error) {
```

Comment:
I believe our designs did not want to generate methods on the types themselves. I would propose we generate a single `backfillAccountID` that handles the type cast and setting the value appropriately.

```go
func backfillAccountID(i interface{}, v string) error {
    switch in := i.(type) {
    case *OperationInput:
       if in.AccountId != nil {
		if !strings.EqualFold(*in.AccountId, v) {
			return fmt.Errorf("error backfilling account id")
		}
		return nil
	}
	in.AccountId = &v
	return nil
    default:
        return fmt.Errorf("unsupported operation")
    }
}
```

Similar feedback for the other member update functions.